In [14]:
import numpy as np
import scipy.stats as st
import matplotlib.pyplot as plt
import seaborn as sns

In [15]:
# A teacher claims that the mean score of students in his class is greater
# than 82 with a standard deviation of 20. If a sample of 81 students was
# selected with a mean score of 90.

p_mew = 82
sample_mean = 90
alpha = 0.05
sigma = 20
sample_no =81

# Calculate the z-value for the given significance level (right-tailed test)
z_value = st.norm.ppf(1 - alpha)

# Calculate the test statistic (z_cal)
z_cal = (sample_mean - p_mew) / (sigma / np.sqrt(sample_no))

# Print results
print("Critical z-value:", z_value)
print("Calculated z:", z_cal)

# Decision
if z_cal > z_value:
    print("Reject H0: The mean score is significantly greater than 82.")
else:
    print("Fail to reject H0: Not enough evidence that the mean score is greater than 82.")

Critical z-value: 1.6448536269514722
Calculated z: 3.5999999999999996
Reject H0: The mean score is significantly greater than 82.


In [16]:
z_value =st.norm.ppf(1-0.05)
z_value

np.float64(1.6448536269514722)

In [17]:
z_cal = (sample_mean - p_mew)/(sigma/np.sqrt(81))
z_cal

np.float64(3.5999999999999996)

In [18]:
if z_value> z_cal: print("H0 is righr")
if z_value <z_cal: print("Ha is right")

Ha is right


In [19]:
# Scenario: Imagine you work for an e-commerce company, and your team is responsible for analyzing
# customer purchase data. You want to determine whether a new website design has led to a significant
# increase in the average purchase amount compared to the old design.

# Data: You have collected data from a random sample of 30 customers who made purchases on the old
# website design and 30 customers who made purchases on the new website design. You have the sample means,
# sample standard deviations=2.5, and sample sizes for both groups.

old = 45.2,42.8,38.9,43.5,41.0,44.6,40.5,42.7,39.8,41.4,44.3,39.7,42.1,40.6,43.0,42.2,41.5,39.6,44.0,38.7,43.9,42.0,41.9,42.8,43.7,41.3,40.9,42.5,41.6,50.6

new = 48.5,49.1,50.2,47.8,48.7,49.9,48.0,50.5,49.8,49.6,48.2,48.9,49.7,50.3,49.4,50.1,48.6,48.3,49.0,50.0,48.4,49.3,49.5,48.8,50.6,50.4,48.1,49.2,50.7,50.8


In [20]:
old_data = np.array(old)
new_data = np.array(new)
len(old),len(new)

(30, 30)

In [21]:
alpha =0.05
sigma_std = 2.5
no_of_sample = 30
new_mean = np.mean(new_data)
old_mean = np.mean(old_data)
print(new_mean,old_mean)



49.346666666666664 42.24333333333333


In [22]:
z_table = st.norm.pdf(1-0.05)
z_table

np.float64(0.254059056469189)

In [23]:
z_calculation = (new_mean-old_mean)/sigma_std/(np.sqrt(no_of_sample))
z_calculation

np.float64(0.5187541200193372)

In [24]:
if z_table > z_calculation:
    print("h0 IS RIGHT")
if z_table < z_calculation:
    print("Ha is right")

Ha is right


In [ ]:
# One-Sample Z-Test: Test if mean of new_data > 48

mu = 48  # Hypothesized mean
sample_mean = new_mean
sample_std = np.std(new_data, ddof=0)
n = no_of_sample

z_stat = (sample_mean - mu) / (sample_std / np.sqrt(n))
p_value = 1 - st.norm.cdf(z_stat)  # right-tailed

print("Z-statistic:", z_stat)
print("p-value:", p_value)

In [ ]:
# Two-Sample Z-Test: Test if mean of new_data > mean of old_data

# Calculate sample means and standard deviations
mean_old = np.mean(old_data)
mean_new = np.mean(new_data)
std_old = np.std(old_data, ddof=0)
std_new = np.std(new_data, ddof=0)
n_old = len(old_data)
n_new = len(new_data)

# Standard error for the difference in means
se = np.sqrt((std_old**2 / n_old) + (std_new**2 / n_new))

# Z-statistic
z_stat_2samp = (mean_new - mean_old) / se

# p-value (right-tailed)
p_value_2samp = 1 - st.norm.cdf(z_stat_2samp)

print("Two-sample Z-statistic:", z_stat_2samp)
print("p-value:", p_value_2samp)

# Understanding the Z-Test

The **z-test** is a statistical method used to determine whether there is a significant difference between sample data and a population, or between two samples. It is commonly used when the population variance is known and the sample size is large (typically n > 30).

---

## When to Use a Z-Test?

- **Sample size is large** (n > 30)
- **Population standard deviation (σ) is known**
- Data is **normally distributed** or sample size is large enough for the Central Limit Theorem to apply

---

## Types of Z-Tests

1. **One-Sample Z-Test**
    - Compares the mean of a single sample to a known population mean.

2. **Two-Sample Z-Test**
    - Compares the means of two independent samples.

---

## Steps in Performing a Z-Test

1. **State the Hypotheses**
    - Null hypothesis (H₀): No difference (e.g., sample mean = population mean)
    - Alternative hypothesis (H₁): There is a difference

2. **Calculate the Z-Statistic**

    \[
    z = {\bar{x} - \mu}{\sigma / \sqrt{n}}
    \]

    **Where:**

    - \( \bar{x} \): Sample mean  
    - \( \mu \): Population mean  
    - \( \sigma \): Population standard deviation  
    - \( n \): Sample size

3. **Find the Critical Value**
    - Based on the chosen significance level (α, e.g., 0.05), use the z-table to find the critical z-value.
    4. **Make a Decision**
        - **If the calculated z is greater than the critical z (for a right-tailed test):**  
          **Reject H₀** (There is enough evidence to support the alternative hypothesis).
        - **If the calculated z is less than or equal to the critical z:**  
          **Fail to reject H₀** (There is not enough evidence to support the alternative hypothesis).

---

## Example

Suppose a teacher claims the average score is greater than 82. A sample of 81 students has a mean score of 90, and the population standard deviation is 20.

- **Null hypothesis (H₀):** μ = 82
- **Alternative hypothesis (H₁):** μ > 82

Calculate z:

\[
z = \frac{90 - 82}{20 / \sqrt{81}} = 3.6
\]

If the critical z-value at α = 0.05 is 1.645, since 3.6 > 1.645, we **reject H₀**.

---

## Key Points

- Z-test helps compare means using standard deviation and sample size.
- Use when population variance is known and sample size is large.
- Compare calculated z to critical z to make a decision.

---

> **Tip:** For small samples or unknown population variance, use the **t-test** instead!

Z-statistic: 8.465553339819863
p-value: 0.0


Two-sample Z-statistic: 16.010869375788996
p-value: 0.0
